pkg/operator/ceph/cluster/osd/osd.go

```diff
@@ -270,6 +270,64 @@ func (c *Cluster) startProvisioning(config *provisionConfig) {
 			}
 		}
 	}
+
+	// Deploying test based on the storageclassDeviceSet
+	// Todo: Replace the below logic to deploy osd pods over pvcs
+	for setIndex, storageClassDeviceSet := range c.DesiredStorage.StorageClassDeviceSets {
+		for i, pvcTemplate := range storageClassDeviceSet.VolumeClaimTemplates {
+			pvcStorageClassDeviceSetPVCId := fmt.Sprintf("%s-%v-%v", storageClassDeviceSet.Name, setIndex, i)
+			pvcStorageClassDeviceSetPVCIdLabelSelector := fmt.Sprintf("rook.ceph.io/StorageClassDeviceSetPVCId=%s", pvcStorageClassDeviceSetPVCId)
+			pvcLables := map[string]string{
+				"rook.ceph.io/storageClassDeviceSet":      storageClassDeviceSet.Name,
+				"rook.ceph.io/storageClassDeviceSetIndex": fmt.Sprintf("%v", setIndex),
+				"rook.ceph.io/pvcIndex":                   fmt.Sprintf("%v", i),
+				"rook.ceph.io/StorageClassDeviceSetPVCId": pvcStorageClassDeviceSetPVCId,
+			}
+
+			// pvc naming format <storageClassDeviceSetName>-<SetNumber>-<PVCIndex>
+			pvcGenerateName := pvcStorageClassDeviceSetPVCId + "-"
+
+			// Add pvcTemplate name to pvc name. i.e <storageClassDeviceSetName>-<SetNumber>-<PVCIndex>-<pvcTemplateName>
```

Comment:
Since we can use this as a label to query for the pvc, what if we just give the generated name a simple prefix such as `rook-ceph`? Seems like we don't need such a complex name for the PVC.